### PR TITLE
Handle json syntax error when opening a pipeline

### LINKED ISSUE #1226: Trying to load invalid pipeline can cause data loss
**Describe the issue**
Create a pipeline and add a node with multiple environment variables
Delete one of the environment variables
Then, right-click the pipeline file, and open it with the editor (not the pipeline editor)
Remove the extra env var that has the value `null` but doesn't remove the coma from the previous env var. 

Now save and try to open on the pipeline editor

No message is displayed for the user, and an empty editor is shown

Two main things:
- The user should be notified that something went wrong when loading the pipeline
- We should not open an empty editor but cancel/close the open to avoid the user being able to save an empty pipeline and lose all previous values from the file.

### PR DESCRIPTION
**Pull request by karlaspuldaro**
Fixes #1226 

When the pipeline JSON file is broken, when trying to open the file on a pipeline editor, the user now sees the dialog:
image
The pipeline tab is closed after pressing the OK button  (avoiding the previous behaviour of displaying a blank pipeline and risking user to save that state and losing all data).

Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: packages/pipeline-editor/src/PipelineEditorWidget.tsx [hunk 1]
```diff
@@ -920,8 +920,19 @@ export class PipelineEditor extends React.Component<
...
   async handleOpenPipeline(): Promise<void> {
     this.widgetContext.ready.then(async () => {
-      let pipelineJson: any = this.widgetContext.model.toJSON();
-      if (pipelineJson == null) {
+      let pipelineJson: any = null;
+
+      try {
+        pipelineJson = this.widgetContext.model.toJSON();
+      } catch (error) {
+        if (error instanceof SyntaxError) {
```
Comment (kevin-bates) [inline packages/pipeline-editor/src/PipelineEditorWidget.tsx:928]: Is this a wide-enough net?  Could any other error could get us into the same data loss situation or is the fact that such errors would now be ignored prevent that from happening?

Comment (lresende) [inline packages/pipeline-editor/src/PipelineEditorWidget.tsx:928]: It's ok if we want to have specific handling for SyntaxError, but we should not restrict to only SyntaxError? If there is an error that is not recoverable, don't open the pipeline and let the user know?

File: packages/ui-components/src/RequestErrors.tsx [hunk 2]
```diff
@@ -94,4 +94,23 @@ export class RequestErrors {
...
+
+  /**
+   * Displays an error dialog containing a syntax error.
+   *
+   * @param fileType - the type of file that was being accessed when
+   * the error occurred
+   *
+   * @returns A promise that resolves with whether the dialog was accepted.
+   */
+  static syntaxError(
```
Comment (lresende) [inline packages/ui-components/src/RequestErrors.tsx:106]: this is not a request error... maybe just make this a utility function on PipelineEditorWidget? or make a more generic error class and make this part of a new class like RuntimeError or something like that...

File: packages/pipeline-editor/src/PipelineEditorWidget.tsx (hunk 1 above)
Comment (karlaspuldaro) [reply packages/pipeline-editor/src/PipelineEditorWidget.tsx:928]: That's a good point, I'll remove the condition

Comment (karlaspuldaro) [reply packages/pipeline-editor/src/PipelineEditorWidget.tsx:928]: I agree. 

> don't open the pipeline and let the user know

In that case we open the pipeline editor only to immediately display the dialog to let the user know what's wrong, then clicking OK will close the tab. Is this a valid behaviour?

**Review by ajbozarth — APPROVED** [review]

**Comment by karlaspuldaro** [comment]
The dialog message:
image

**Review by lresende — APPROVED** [review]

**Pull request merged**